test/backup-restore/mzcompose.py

```diff
@@ -0,0 +1,162 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+
+from materialize.mzcompose.composition import Composition
+from materialize.mzcompose.service import Service
+from materialize.mzcompose.services.cockroach import Cockroach
+from materialize.mzcompose.services.materialized import Materialized
+from materialize.mzcompose.services.minio import Mc, Minio
+from materialize.mzcompose.services.testdrive import Testdrive
+
+SERVICES = [
+    Cockroach(setup_materialize=True),
+    Minio(setup_materialize=True),
+    Mc(),
+    Materialized(external_minio=True, external_cockroach=True, sanity_restart=False),
+    Testdrive(no_reset=True),
+    Service(
+        name="persistcli",
+        config={
+            # TODO: depends!
+            "mzbuild": "jobs"
+        },
+    ),
+]
+
+
+def workflow_default(c: Composition) -> None:
+    # TODO: extract common substrings
+
+    # Enable versioning for the Persist bucket
+    c.up("minio")
+    c.run(
+        "mc",
+        "alias",
+        "set",
+        "persist",
+        "http://minio:9000/",
+        "minioadmin",
+        "minioadmin",
+    )
+    c.run("mc", "version", "enable", "persist/persist")
+    blob_uri = "s3://minioadmin:minioadmin@persist/persist?endpoint=http://minio:9000/&region=minio"
```

Comment:
AFAICT this would always be worthwhile for minio-based tests, but it does involve yet another container in the mix, which the existing minio tool goes out of its way to avoid.

Reply:
I am going to do this in Platform checks and other frameworks, so most minio-based tests will start using versioning .